### PR TITLE
Tracker: Refactor ViewOptions

### PR DESCRIPTION
**Pull request by Dvinyanin**
Signed-off-by: Dvinyanin Alexandr <dvinyanin.alexandr@gmail.com>

# Contribution checklist

## Brief description
Screenshot from 2022-07-08 10-28-11

## Checklist

* [ ] - UI test added to added/changed functionality?
* [x] - Screenshot is added to PR if applicable ?
* [x] - Does the code work? Check whether function and logic are correct.
* [ ] - Does Changelog.md is updated with changes?
* [x] - Does the translations are up to date?
* [x] - Does it well tested?
* [x] - Tested for Chrome.
* [ ] - Tested for Safari.
* [x] - Go through the changed code looking for typos, TODOs, commented LOCs, debugging pieces of code, etc.
* [x] - Rebase your branch onto master and upstream branch
* [x] - Is there any redundant or duplicate code?
* [x] - Are required links are linked to PR?
* [x] - Does new code is well documented ?

## Related issues

A list of closed updated issues
https://front.hc.engineering/workbench/platform/tracker/TSK-262

### CODE REVIEW

File: plugins/view-resources/src/components/ViewOptionsButton.svelte [hunk 1]
```diff
@@ -13,37 +13,31 @@
...
-          completedIssuesPeriod: IssuesDateModificationPeriod.All,
-          shouldShowEmptyGroups: false,
-          shouldShowSubIssues: false
-        }
-  }
+  export let config: ViewOptionModel[]
+  export let viewOptionsKey: string
 
-  $: $viewOptionsStore = viewOptions
+  $: $viewOptionsStore = config.reduce(
```
Comment (haiodo) [inline plugins/view-resources/src/components/ViewOptionsButton.svelte]: I'm confused, Svelte, will subscribe for viewOptionsStore and create local $viewOptionsStore, why you update it? if you plan to update store value, pleas use viewOptionsStore.set() instead.

Comment (Dvinyanin) [reply plugins/view-resources/src/components/ViewOptionsButton.svelte]: Is this misuse of store?
https://github.com/hcengineering/anticrm/blob/190954e74792b24263ddf340b9202180c3ef559d/plugins/view-resources/src/components/filter/FilterBar.svelte#L89

Comment (haiodo) [reply plugins/view-resources/src/components/ViewOptionsButton.svelte]: Ok I was wrong, checked generated code.

```
		if (saved !== null) {
			set_store_value(filterStore, $filterStore = JSON.parse(saved), $filterStore);
		} else {
			set_store_value(filterStore, $filterStore = [], $filterStore);
		}
```
It indeed update store.

**Review by haiodo — APPROVED** [review]

File: plugins/view-resources/src/components/ViewOptionsButton.svelte (hunk 1 above)
Comment (Dvinyanin) [reply plugins/view-resources/src/components/ViewOptionsButton.svelte]: I already change it to update store explicitly

**Pull request merged**